Source/WTF/wtf/URL.cpp

```diff
@@ -531,20 +531,21 @@ template<typename StringType>
 static String percentEncodeCharacters(const StringType& input, bool(*shouldEncode)(UChar))
 {
     auto encode = [shouldEncode] (const StringType& input) {
-        CString utf8 = input.utf8();
-        auto* data = utf8.data();
-        StringBuilder builder;
-        auto length = utf8.length();
-        for (unsigned j = 0; j < length; j++) {
-            auto c = data[j];
-            if (shouldEncode(c)) {
-                builder.append('%');
-                builder.append(upperNibbleToASCIIHexDigit(c));
-                builder.append(lowerNibbleToASCIIHexDigit(c));
-            } else
-                builder.append(c);
-        }
-        return builder.toString();
+        auto result = input.tryGetUtf8ForRange([&](Span<const char> span) -> String {
+            StringBuilder builder;
+            for (unsigned j = 0; j < span.size(); j++) {
+                auto c = span[j];
+                if (shouldEncode(c)) {
+                    builder.append('%');
+                    builder.append(upperNibbleToASCIIHexDigit(c));
+                    builder.append(lowerNibbleToASCIIHexDigit(c));
+                } else
+                    builder.append(c);
+            }
+            return builder.toString();
+        }, 0, input.length());
+        RELEASE_ASSERT(result);
```

Comment:
Why did we need this to be a release assertion? Is there something dangerous about a null `String`?

Reply:
This code is aligning to the old `utf8()` call here.